src/BuiltinExtensions/ComfyUIBackend/ExtraNodes/SwarmComfyCommon/SwarmSaveImageWS.py

```diff
@@ -22,19 +26,44 @@ def INPUT_TYPES(s):
     OUTPUT_NODE = True
     DESCRIPTION = "Acts like a special version of 'SaveImage' that doesn't actual save to disk, instead it sends directly over websocket. This is intended so that SwarmUI can save the image itself rather than having Comfy's Core save it."
 
-    def save_images(self, images):
+    def save_images(self, images, bit_depth):
         pbar = comfy.utils.ProgressBar(SPECIAL_ID)
         step = 0
         for image in images:
-            i = 255.0 * image.cpu().numpy()
-            img = Image.fromarray(np.clip(i, 0, 255).astype(np.uint8))
+            img_np = image.cpu().numpy()
+            if bit_depth == "16bit":
+                img_np = np.clip(img_np * 65535.0, 0, 65535).astype(np.uint16)
+                img = self.convert_opencv_to_pil(img_np)
+            else:
+                i = np.clip(img_np * 255.0, 0, 255).astype(np.uint8)
```

Comment:
I again must ask, why was this code altered?

Reply:
I don't think I understand the question.
Why did I write this code, as in what is it doing now? Or why is it written this specific way, rather than

```
if bit_depth == "16bit":
    img_np = np.clip(img_np * 65535.0, 0, 65535).astype(np.uint16)
    img = self.convert_opencv_to_pil(img_np)
else:
    i = 255.0 * image.cpu().numpy()
    img = Image.fromarray(np.clip(i, 0, 255).astype(np.uint8))
```

If the first, the code was modified to handle the expanded value range (0-65535) and data type (uint16)

If the second, maybe just not as much elegance in writing pythonic code.

```
img_np = image.cpu().numpy()
if bit_depth == "16bit":
    img_np = np.clip(img_np * 65535.0, 0, 65535).astype(np.uint16)
    img = self.convert_opencv_to_pil(img_np)
else:
    img = Image.fromarray(np.clip(img_np * 255.0, 0, 255).astype(np.uint8))
```

If the critique is on something else I'm not seeing, I'd appreciate the pointer on what is it that I'm missing

Reply:
the code for the 8bit section is different than it was before the PR. The red lines don't match the green lines, even though the only expected change is those lines get tabbed forward into the else block. Why is there a difference?

Reply:
Ah, I see.

Before it was:
```
        for image in images:
            i = 255.0 * image.cpu().numpy()
            img = Image.fromarray(np.clip(i, 0, 255).astype(np.uint8))
            pbar.update_absolute(step, SPECIAL_ID, ("PNG", img, None))
            step += 1
```
I noticed that this was grabbing the output of the tensor as a numpy array `image.cpu().numpy()` and scaling it to the 8bit color range. I was going to have to do the same operation for scaling it to 16bits, so it made sense in my mind to extract the numpy array into it's own variable, so I could just repeat the scaling with the appropriate ranges for 16bits.
I suppose another way I could have done the same thing is:

```
if bit_depth == "16bit":
    i = 65535.0 * image.cpu().numpy()
    img = self.convert_opencv_to_pil(np.clip(i, 0, 65535).astype(np.uint16))
else:
    i = 255.0 * image.cpu().numpy()
    img = Image.fromarray(np.clip(i, 0, 255).astype(np.uint8))
```
and this would have kept the syntax for 8bits same as before, and probably cleaner to review too. Is this what you're referring to? If so, I didn't really see this option until now, to be honest.

Reply:
yes

Reply:
thank you for your patience. I'll fix this and push it up